kedro/__init__.py

```diff
@@ -9,11 +9,16 @@
 __version__ = "0.18.13"
 
 
+class KedroDeprecationWarning(DeprecationWarning):
+    """Custom class for warnings about deprecated Kedro features."""
+
+
+warnings.filterwarnings("default", category=KedroDeprecationWarning)
```

Comment:
Since the global warning registry overrides any CLI option https://discuss.python.org/t/how-to-silence-pkg-resources-warnings/28629/7?u=astrojuanlu, could we have this behind a conditional?

```suggestion
if not sys.warnoptions:
    warnings.filterwarnings("default", category=KedroDeprecationWarning)
```

see also https://docs.python.org/3.7/library/warnings.html#overriding-the-default-filter

Reply:
(Actually there's a `if not sys.warnoptions` below already)

Reply:
Done! Also changed it to a `simplefilter`, since not using `message` anyway.